src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,35 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save actor to DB. Actor=" + actor, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```

Comment:
```suggestion
        }
```

Reply:
Use try with finally, where you use transaction and try with resources with Read operations. - From Common Mistakes.
We need to close our sessions.